sklearn/preprocessing/tests/test_data.py

```diff
@@ -822,14 +824,9 @@ def test_scale_sparse_with_mean_raise_exception():
 
 def test_scale_input_finiteness_validation():
     # Check if non finite inputs raise ValueError
-    X = [[np.nan, 5, 6, 7, 8]]
-    assert_raises_regex(ValueError,
-                        "Input contains NaN, infinity or a value too large",
-                        scale, X)
-
     X = [[np.inf, 5, 6, 7, 8]]
     assert_raises_regex(ValueError,
-                        "Input contains NaN, infinity or a value too large",
+                        "Input contains infinity or a value too large",
```

Comment:
This actually a weird message: what is a "value too large" if it's not infinity?

Reply:
If the values vary too much and computing the scale is not numerically possible (overflow)? If so we should add a specific test for this.

Reply:
This is not the reason:

```python
>>> from sklearn.preprocessing import scale
>>> import numpy as np
>>> data = np.array([np.finfo(np.float32).max, np.finfo(np.float32).min])
>>> scale(data)
/home/ogrisel/.virtualenvs/py36/lib/python3.6/site-packages/numpy/core/_methods.py:116: RuntimeWarning: overflow encountered in multiply
  x = um.multiply(x, x, out=x)
array([ 0., -0.], dtype=float32)
```

So I would just change the message to "Input contains infinity".

Reply:
Uhm this is the error message of `check_array` actually. Changing this part would require to touch quite a lot of tests not really related. Would it be wised to make it inside another PR.